File: Data Structures & Algorithms/same-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isSameTree(TreeNode p, TreeNode q) {
        if(p == null && q == null){
            return true;
        }
        else if (p != null && q == null){
            return false;
        }
        else if(p == null && q != null){
            return false;
        }
        else{
            if(p.val != q.val){
                return false;
            }
           
            else{
                boolean ls = isSameTree(p.left,q.left);
                boolean rs = isSameTree(p.right,q.right);
                boolean yes = ls && rs;
               return yes;
               
            }
        }
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        self.ans = []
        
        def bfs(root):
            queue = deque([root])
            while len(queue) > 0:
                if not root:
                    return self.ans
                    
                n = len(queue)
                
                for i in range(n):
                    x = queue.popleft()

                    if i == n - 1:
                        self.ans.append(x.val)
                    
                    if x.left:
                        queue.append(x.left)

                    if x.right:
                        queue.append(x.right)
        bfs(root)

        return self.ans
